### PR TITLE
rpmbuild: improvements for building with rbd

### PR DESCRIPTION
**Pull request by guptsanj**
Signed-off-by: Sanjeev Gupta [guptsanj@us.ibm.com](https://github.com/ceph/spdk/compare/ceph-nvmeof...guptsanj:guptsanj/guptsanj@us.ibm.com)

Depends on https://github.com/ceph/spdk/pull/7

```
 Cherry-pick rbd support from spdk upstream.

   Author: Mykola Golub <mykola.golub@clyso.com>
   Date:   Sun Jan 16 17:10:31 2022 +0000

    rpmbuild: improvements for building with rbd

    Add librbd and librados install and build dependencies when "with-rbd"
    config option is set.

    This will be used by the Ceph NVMe-oF gateway.

    Signed-off-by: Mykola Golub <mykola.golub@clyso.com>
    Change-Id: I1534e224cdfc166061a166ceef18261eef2d8cd8
    Reviewed-on: https://review.spdk.io/gerrit/c/spdk/spdk/+/11135
    Reviewed-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
    Reviewed-by: Jim Harris <james.r.harris@intel.com>
    Tested-by: SPDK CI Jenkins <sys_sgci@intel.com>
```

### CODE REVIEW

File: rpmbuild/rpm.sh [hunk 1]
```diff
@@ -105,6 +105,20 @@ build_macros() {
...
+		build_requirements=${build_requirements:+$build_requirements, }"librados-devel, librbd-devel"
+	fi
+
+	if get_config libdir has-arg; then
+		macros+=(-D "libdir $(get_config libdir print)")
+	fi
+
+	if get_config with-vfio-user; then
+		macros+=(-D "vfio_user 1")
+	fi
```
Comment (epuertat) [inline rpmbuild/rpm.sh:120]: Are these needed? When I resolved this I noticed this came from an unrelated PR (but haven't 'built' it, so I'm not sure if these deps are really required)

File: rpmbuild/spdk.spec [hunk 2]
```diff
@@ -8,6 +8,9 @@
 %{!?requirements:%define requirements 0}
 %{!?build_requirements:%define build_requirements 0}
 %{!?shared:%define shared 0}
+%{!?rbd:%define rbd 0}
+%{!?libdir:%define libdir /usr/local/lib}
+%{!?vfio_user:%define vfio_user 0}
```
Comment (epuertat) [inline rpmbuild/spdk.spec:13]: Same here.

File: scripts/pkgdep.sh [hunk 3]
```diff
@@ -34,11 +36,12 @@ INSTALL_CRYPTO=false
 INSTALL_DEV_TOOLS=false
 INSTALL_PMEM=false
 INSTALL_FUSE=false
+INSTALL_RBD=false
 INSTALL_RDMA=false
 INSTALL_DOCS=false
 INSTALL_LIBURING=false
 
-while getopts 'abdfhipru-:' optchar; do
+while getopts 'abdfhipruDR-:' optchar; do
```
Comment (epuertat) [inline scripts/pkgdep.sh]:
```suggestion
while getopts 'abdfhipruR-:' optchar; do
```
That's also in the other PR

**Review by epuertat — APPROVED** [review]
LGMT, except for the few lines that seem to come from other commits.

BTW, it's ok to have multi-commit PRs, especially if the commits are related and dependent on each other. So we can simply work on this one and discard the other.

File: rpmbuild/rpm.sh (hunk 1 above)
Comment (guptsanj) [reply rpmbuild/rpm.sh:120]: Will test the spdk build after removing these lines 
Working on setting up a ceph cluster where I can test the gateway

File: rpmbuild/spdk.spec (hunk 2 above)
Comment (guptsanj) [reply rpmbuild/spdk.spec:13]: Will test the spdk build after removing these lines 
Working on setting up a ceph cluster where I can test the gateway

**Comment by idryomov** [comment]
Included in SPDK 23.01

**Pull request closed without merging**